app/Http/Controllers/V2/Dashboard/CountryDataController.php

```diff
@@ -0,0 +1,108 @@
+
+        // Extract the bounding box coordinates
+        $coordinates = $geoJson->coordinates[0];
+
+        // Get the country name
+        $countryName = $countryData->country;
+
+        // Construct the bbox data in the specified format
+        $countryBbox = [
+            $countryName,
+            [$coordinates[0][0], $coordinates[0][1], $coordinates[2][0], $coordinates[2][1]],
+        ];
+
+        return response()->json(['bbox' => $countryBbox]);
+    }
+
+    public function getPolygonData(string $uuid)
+    {
+        $sitePolygon = SitePolygon::where('poly_id', $uuid)->first();
+
+        if (! $sitePolygon) {
+            return response()->json(['error' => 'Polygon not found'], 404);
+        }
+
+        $project = $sitePolygon->project()->first();
+
+        if (! $project) {
+            Log::error("Project not found for site polygon with ID: $sitePolygon->id");
+        }
+
+        $site = $sitePolygon->site()->first();
+
+        if(! $site) {
+            Log::error("Site not found for site polygon with ID: $sitePolygon->id");
+
+        }
+
+        $data = [
+          ['key' => 'poly_name', 'title' => 'title', 'value' => $sitePolygon->poly_name ?? null],
+          ['key' => 'project_name', 'title' => 'Project', 'value' => $project->name ?? null],
+          ['key' => 'site_name', 'title' => 'Site', 'value' => $site?->name ?? null],
+          ['key' => 'num_trees', 'title' => 'Number of trees', 'value' => $sitePolygon->num_trees ?? null],
+          ['key' => 'plantstart', 'title' => 'Plant Start Date', 'value' => $sitePolygon->plantstart ?? null],
+          ['key' => 'status', 'title' => 'Status', 'value' => $sitePolygon->status ?? null],
+
+        ];
+
+        return response()->json(['data' => $data]);
+    }
+
+    public function getProjectData(string $uuid)
+    {
+        try {
+            $project = Project::isUuid($uuid)->first();
+
+            if (! $project) {
+                Log::error("Project not found for project with UUID: $uuid");
+                return response()->json(['error' => 'Project not found'], 404);
+            }
+            $countSitePolygons = $project?->getTotalSitePolygons() ?? 0;
+
+            $organization = $project->organisation()->first();
+            if (! $organization) {
+                Log::error("Organization not found for project with ID: $project->id");
+            }
+
+            $country = $project ? WorldCountryGeneralized::where('iso', $project->country)->first() : null;
+            $data = [
+              ['key' => 'project_name', 'title' => 'title', 'value' => $project?->name ?? null],
+              ['key' => 'country', 'title' => 'Country', 'value' => $country?->country ?? null],
+              ['key' => 'polygon_counts', 'title' => 'No. of Site - Polygons', 'value' => $countSitePolygons ?? null],
+              ['key' => 'organizations', 'title' => 'Organization', 'value' => $organization->name ?? null],
```

Comment:
Since there's no early exit on a null organization above, this needs the safe navigation operator, and again here `?? null` isn't doing anything useful:
```
'value' => $organization?->name],
```

Reply:
fixed